### ISSUE TITLE
OIDC device flow: visual indication that notebook is waiting

### ISSUE THREAD

**Issue by soxofaan** [Open-EO/openeo-python-client#237]
When doing the OIDC device flow in a notebook (or even as a regular script), the `con.authenticate_oidc()` call (or related) will print something like 
```
To authenticate: visit https://aai.egi.eu/oidc/device and enter the user code 'F0oB6r'.
```
and then do noting (except waiting).

It would be cool if there was a more visual indication that the client code is waiting for the user to do something: e.g. some kind of spinner, progress bar (based on waiting timeout), ...

**Comment by soxofaan** [comment]
problem with current implementation: output  text widget in jupyter notebook is now refreshed every 1, which makes it hard/impossible to select the url or code for copy pasting, which is a usability regression

**Comment by soxofaan** [comment]
Also to do: progress bar should be cleaned up (or a new line should be added), now it looks like this:
```
To authenticate: visit https://aai.egi.eu/oidc/device and enter the user code '.....'.
Authorized successfully.   ] Waiting for authentication.             
Authenticated using device code flow.
```

**Comment by soxofaan** [comment]
implemented in #420
